source/fuzz/fuzzer_util.cpp

```diff
@@ -1925,6 +1925,15 @@ bool NewTerminatorPreservesDominationRules(opt::IRContext* ir_context,
   return true;
 }
 
+opt::Module::iterator GetFunctionIterator(opt::IRContext* ir_context,
+                                          uint32_t function_id) {
+  for (auto iter = ir_context->module()->begin();
+       iter != ir_context->module()->end(); ++iter) {
+    if ((*iter).result_id() == function_id) return iter;
+  }
+  return ir_context->module()->end();
+}
```

Comment:
It might be better to use `std::find_if` instead of implementing it manually.

Reply:
@sliu-UIUC This is a good idea; can you also do it in a follow-up PR?

Reply:
> It might be better to use `std::find_if` instead of implementing it manually.

Hi Vasyl, I rarely used `std::find_if` so I don't know how to use it that much. But when I do try using `find_if` in the function
iterator, it seemed like the built-in `find_if` doesn't work well with `opt::module::iterator` type and kept throwing out errors.

Reply:
On catch with find_if is that the lambda takes a value an iterator points to, not the iterator itself. In your case, it will be
```cpp
std::find_if(ir_context->module()->begin(), ir_context->module()->end(), [result_id](const opt::Function& f) {
  return f.result_id() == result_id;
})
```
See https://en.cppreference.com/w/cpp/algorithm/find.

Reply:
> One catch with find_if is that the lambda takes a value an iterator points to, not the iterator itself. In your case, it will be
> 
> ```c++
> std::find_if(ir_context->module()->begin(), ir_context->module()->end(), [result_id](const opt::Function& f) {
>   return f.result_id() == result_id;
> })
> ```
> 
> See https://en.cppreference.com/w/cpp/algorithm/find.

Got it. Thanks for the feedback!